src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -0,0 +1,144 @@
+                    logger.debug("Ably BATCH call -Thread(" + Thread.currentThread().getName() + ")" + "Num Records:" + value.size());
+                    try {
+                        sendBatches(new BatchSpec(Set.of(key), value));
+                    } catch (AblyException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Function to group the Ably Message objects by channel Name.
+     * @param sinkRecords
+     * @return
+     */
+    public Map<String, List<Message>> groupMessagesByChannel(List<SinkRecord> sinkRecords) {
+
+        HashMap<String, List<Message>> channelNameToMessagesMap = new HashMap();
+
+        for(SinkRecord record: sinkRecords) {
+            try {
+                if (!shouldSkip(record)) {
+                    final String channelName = channelSinkMapping.getChannelName(record);
+                    final Message message = messageSinkMapping.getMessage(record);
+
+                    List<Message> messages;
+                    if (channelNameToMessagesMap.containsKey(channelName)) {
+                        // just retrieve the list and add to it.
+                        messages = channelNameToMessagesMap.get(channelName);
+                    } else {
+                        // Create a new Set.
+                        messages = new ArrayList<>();
+                    }
+                    messages.add(message);
+                    channelNameToMessagesMap.put(channelName, messages);
+                }
+            } catch(Exception e) {
+                logger.error("Error transforming sink record", e);
+            }
+        }
+
+        return channelNameToMessagesMap;
+    }
+
+    protected boolean shouldSkip(SinkRecord record) {
+        final boolean skipOnKeyAbsence = connectorConfig.getBoolean(SKIP_ON_KEY_ABSENCE);
+
+        if (skipOnKeyAbsence) {
+            final String messageConfig = connectorConfig.getString(MESSAGE_CONFIG);
+            final String channelConfig = connectorConfig.getString(CHANNEL_CONFIG);
+            final ConfigValueEvaluator.Result messageResult = configValueEvaluator.evaluate(record, messageConfig, true);
+            final ConfigValueEvaluator.Result channelResult = configValueEvaluator.evaluate(record, channelConfig, true);
+
+            if (messageResult.shouldSkip() || channelResult.shouldSkip()) {
+                logger.warn("Skipping record as record key is not available in a record where the config for either" +
+                        " 'message.name' or 'channel' is configured to use #{key} as placeholders {}", record);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Function that uses the Ably REST API
+     * to send records in batches.
+     * @param batch
+     * @throws AblyException
+     */
+    public void sendBatches(final BatchSpec batch) throws AblyException {
```

Comment:
This function should take `List<BatchSpec>` as per the original gist, so that you don't need to make separate batch submissions per channel.